packages/pluggableWidgets/video-player-web/src/components/SizeContainer.tsx

```diff
@@ -21,59 +23,67 @@ export interface SizeProps extends Dimensions {
-                <div className={classNames("size-box-inner", this.props.classNameInner)} style={absoluteStyle}>
-                    {this.props.children}
-                </div>
+                <div className={classNames("size-box-inner", this.props.classNameInner)}>{this.props.children}</div>
             </div>
         );
     }
 
-    private getHeight(
-        heightUnit: HeightUnitType,
-        height: number,
-        widthUnit: WidthUnitType,
-        width: number
-    ): CSSProperties {
+    private setDimensions(): CSSProperties {
+        const { widthUnit, width, heightUnit, height, heightAspectRatio } = this.props;
+
         const style: CSSProperties = {};
-        if (heightUnit === "percentageOfWidth") {
-            const ratio = (height / 100) * width;
-            if (widthUnit === "percentage") {
-                style.height = "auto";
-                style.paddingBottom = `${ratio}%`;
-            } else {
-                style.height = `${ratio}px`;
-            }
-        } else if (heightUnit === "pixels") {
-            style.height = `${height}px`;
-        } else if (heightUnit === "percentageOfParent") {
-            style.height = `${height}%`;
-        } else if (heightUnit === "aspectRatio") {
-            style.height = width * 0.5625; // Default is 16:9
+        style.width = widthUnit === "percentage" ? `${width}%` : `${width}px`;
+
+        switch (heightUnit) {
+            case "pixels":
+                style.paddingTop = height;
+                break;
+            case "percentageOfWidth":
+                const actualHeight = (height! / 100) * width;
+
+                if (widthUnit === "percentage") {
+                    style.paddingTop = `${actualHeight}%`;
+                } else {
+                    style.paddingTop = actualHeight;
+                }
+                break;
+            case "percentageOfParent":
+                style.paddingTop = 0;
+                style.height = `${height}%`;
+                break;
+            case "aspectRatio":
+                let ratioHeightFactor = 0;
+
+                switch (heightAspectRatio) {
+                    case "oneByOne":
+                        ratioHeightFactor = 1;
+                        break;
+                    case "fourByThree":
+                        ratioHeightFactor = 3 / 4;
+                        break;
+                    case "threeByTwo":
+                        ratioHeightFactor = 2 / 3;
+                        break;
+                    case "sixteenByNine":
+                        ratioHeightFactor = 9 / 16;
+                        break;
+                    case "TwentyOneByNine":
+                        ratioHeightFactor = 9 / 21;
+                        break;
+                }
+
+                if (widthUnit === "pixels") {
+                    style.paddingTop = width * ratioHeightFactor;
+                } else {
+                    style.paddingTop = `${width * ratioHeightFactor}%`;
```

Comment:
I'm not entirely sure whether this case is correct. If we are using an aspect ratio, why would the unit of the width affect the value of the height? If we want a 21:9 video, I assume the height should always be (9/21)*100 = 42.86% of the width. 

In the if clause you go back to pixels, which is fine but I don't think necessary since it should always be `${ratioHeightFactor}%`. 

But in the else clause, let's say we have width 50% and the container around it is 420pixels. The width is now 210 pixels because we set with to 50%. Then we determine the height with 21:9. In your calculations, that would result in `ratioHeightFactor` being 0.4286. This else case would then mulitply that with the `width`, which would lead to `style.paddingTop` being 0.4286 * 50 = 21.43% (you also see this in your spec test). 

But paddings percentages are relative to the width if I recall correctly (please confirm), so that would mean your height is now 160 * 0.2143 = 45 pixels which is incorrect.

Instead it should stick to 0.4286 as that would result in 160 * 0.4286 = 90 pixels, as you would assume in case of a 21:9 ratio 210 pixels width element.

Reply:
Padding percentages are relative to the width of the **parent** element. That makes the current code valid.

Reply:
Ahh I didn't consider that the width is targeting the _current_ element, whereas the padding top percentage is relative to the parent but targeting the current. Since this function is honestly quite complex (and difficult to follow properly cuz of it), can you add some tests to verify these results? especially the aspect ratios and the different scenarios that can occur due to `widhtUnit`. 

You can do so by extending the current snapshot tests or unit test this function (former is prob easier), I will leave that up to you. I think it's better to have concrete results written down in this case. You don't have to repeat them for every aspect ratio obviously, but picking some values that are easy to verify everything went well (like in my example) is prob nice

Reply:
I agree. Changed it.